src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,71 @@
+package duke.parser;
+
+import duke.command.*;
+
+import duke.exception.DukeException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * The parser of the user command.
+ */
+public class Parser {
+
+    /**
+     * Parses the input command.
+     *
+     * @param fullCommand The full string of user's command.
+     * @return The Command object to react.
+     * @throws DukeException When input is in wrong format.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        if (fullCommand.equals("bye")) {
+            return new ExitCommand();
+        } else if (fullCommand.equals("list")) {
+            return new ListCommand();
+        } else if (fullCommand.startsWith("done")) {
+            char num = fullCommand.charAt(fullCommand.length() - 1);
+            int index = Character.getNumericValue(num);
```

Comment:
Regarding the input processing(for example extracting the description & dates) in **Parser** class, I think you can abstract them away by putting them in the respective **Command** classes to make the Parser class code cleaner. e.g. You can abstract away the input processing logic for "done" and put them in the **DoneCommand** class. Then, you would only need to pass the raw input to the DoneCommand for it to process.

Reply:
Also maybe you could store similar classes such as **Parser, Storage** in one package instead of different packages? Cuz they are like the supporting components for the Duke bot. This way it may be easier to navigate through the directories and find similar classes.